### PR TITLE
adding Event and security section

### PR DESCRIPTION
**Pull request by hbaptiste**
adding events and security section

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -16,6 +16,9 @@ pages:
         - 'Installation': 'developper-documentation/installation.md'
         - 'Architecture': 'developper-documentation/architecture.md'
         - 'Configuration': 'developper-documentation/configuration.md'
+        - 'Events': 'developper-documentation/events.md'
```
Comment (mickaelandrieu) [inline mkdocs.yml]: should be in "Components part" imo

File: doc/developper-documentation/ClassContentEvent.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+is dealing with a request, a common way to listen to them is to use an `event.yml` config file.
+Will try to local event file in the `repository/Config` folder. Bellow is an example of how to
+use the file.
+
+```
+    article.article.render:
+		listeners:
+			- [BackBee\Event\Listener\ArticleListener, onRender]
+
+	social.facebook.prerender:
```
Comment (mickaelandrieu) [inline doc/developper-documentation/ClassContentEvent.md]: care indentation, this sample will produce a Yaml parsing exception error

File: doc/developper-documentation/ClassContentEvent.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+was rendered.
+`*.render`  and `*.prerender` are events triggered by the `Renderer` object. Bellow is the list of
+all the Renderer events.
+
+- *.prerender
+- *.render
+- *.postrender
+
+*prerender* is triggered *before* the content is been rendered. *render* when the
+ content is about te be rendered and *postrender* after the render process. How ever There is no strong difference *render* and *postrender*.
```
Comment (mickaelandrieu) [inline doc/developper-documentation/ClassContentEvent.md]: is about to

File: doc/developper-documentation/ClassContentEvent.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+As in BackBee all the contents are `Doctrine` entities, these doctrine events are also available for all the contents.
+
+- preremove
+- postremove
+- postupdate
+- preupdate
+- update
+- prepersisttheser
```
Comment (mickaelandrieu) [inline doc/developper-documentation/ClassContentEvent.md]: ``prepersist`` imo

File: doc/developper-documentation/ClassContentEvent.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+- postpersist
+- postload
+- onflush
+
+
+## ClassContent Listener
+To listen to a ClassContentEvent we have to create a `Listener` class. By default Backbee
+will looks for Listerner in the repository/Config/Listener folder.
```
Comment (mickaelandrieu) [inline doc/developper-documentation/ClassContentEvent.md]: Listener

File: doc/developper-documentation/ClassContentEvent.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+```
+    <?php
+		namespace BackBee\Event\Listener,
```
Comment (mickaelandrieu) [inline doc/developper-documentation/ClassContentEvent.md]: care indentation, you can use the command ``mkdocs serve`` to take a look to final rendering of this markdown file

File: doc/developper-documentation/ClassContentEvent.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+when a content is created, before and after a content has been saved, when the rendering process
+is started, while the content is been rendered, after the renderer process and so on. these events
+allow us the alter in different ways the contents lifecycle or the rendering process itself.
+
+ClassContent events are similar to custom events. But, as they take place when the main application
+is dealing with a request, a common way to listen to them is to use an `event.yml` config file.
+Will try to local event file in the `repository/Config` folder. Bellow is an example of how to
+use the file.
+
+```
```
Comment (ndufreche) [inline doc/developper-documentation/ClassContentEvent.md]: specify file type

File: doc/developper-documentation/ClassContentEvent.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+- postpersist
+- postload
+- onflush
+
+
+## ClassContent Listener
+To listen to a ClassContentEvent we have to create a `Listener` class. By default Backbee
+will looks for Listerner in the repository/Config/Listener folder.
+
+```
```
Comment (ndufreche) [inline doc/developper-documentation/ClassContentEvent.md]: specify file type

File: doc/developper-documentation/events.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+```
+
+## Event Listener
+`Namespace: BackBee\Event\EventListener`
+
+To listen to an event we must use the `addListener` method of the `EventDispatcher`.
+`addListener` takes three parameters. The first one is the name of the event that will be triggered, the second is an array whose first element
+is a class path and the second is a method. The last parameter is the priority of the event. Listener with higher priority will be triggered first.
+
```
Comment (ndufreche) [inline doc/developper-documentation/events.md]: add ```php

File: doc/developper-documentation/events.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+Notice that the event object has access to the `application` *via* the `EventDispacher` and that the `sayHello` is a *static* method.
+
+## Event Dispatcher
+`Namespace: BackBee\Event\EventListener`
+
+Events are *dispatched* or *triggered* by the `EventDispatcher` by using the method `triggerEvent`
+The sample bellow shows how to listen to an event.
+
+To trigger an event, we must use the `EventDispatcher`. It provides, among others, a `dispatch` method that takes two arguments. A name and an event object.
+
```
Comment (ndufreche) [inline doc/developper-documentation/events.md]: add ```php

File: doc/developper-documentation/events.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+`Namespace: BackBee\Event\Event`
+
+The Backbee event class extend `sfEvent` and allows us to create a generic `Event` object.
+An generic event object has two properties: `target` and `args`;
+The target property represents the target of the event. The `args` is an optional associated map.
+When the `args` property is provided, it's sent to the listeners.
+Custom Event can be created by extending the class `BackBee\Event\Event`.
+
+```
```
Comment (ndufreche) [inline doc/developper-documentation/events.md]: specify file type

**Comment by mickaelandrieu** [comment]
@hbaptiste thank you for your rework, I will get your work and update according to the review.

**Pull request closed without merging**